### PR TITLE
Pull request to merge with development

### PR DESCRIPTION
**Pull request by marcos-medeiros**
# Thanks a lot for taking the time to review our code!

## Done work

- [x] Create event, user and attendance models.

- [x] Create event, user, attendance and session controllers.

- [x] Create a signup page; a user's show page; an event's index, new page, and a show page.

- [x] Create a navigation bar with useful links.

- [x] Create tests for the application.

### CODE REVIEW

File: app/views/events/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<h2>Upcoming Events</h2>
```
Comment (mosaaleb) [inline app/views/events/index.html.erb]: - [ ] **[OPTIONAL]** We can see that we have alot of duplicate code in here. We can create a [partial](https://guides.rubyonrails.org/layouts_and_rendering.html#using-partials) for event with the name `_event.html.erb`.

After moving that data you want to display (without the for loop) to a partial you can use this partials as follows: 
```erb
<%= render @upcoming_events %>
<%= render @past_events %>
```
This is called [collection rendering](https://guides.rubyonrails.org/layouts_and_rendering.html#rendering-collections).

File: app/views/events/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<h1><%= @event.name %></h1>
+<p>Date created: <%= @event.date %></p>
+<p>Description: <%= @event.description %></p>
+<p>Location: <%= @event.location %></p>
+<p>Host: <%= @event.creator.name %>
+
+<p> Guest List </p>
+<ul>
+<% @att_event.each do |a| %>
```
Comment (mosaaleb) [inline app/views/events/show.html.erb]: - [ ] **[OPTIONAL]** We can also loop through `@event.attendees`, just more readable.

File: app/views/events/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  <li><%= a.user.name %></li>
+<% end %>
+</ul>
+
+<%= button_to 'Join event', attendances_new_path(id: @event.id), method: :post %>
```
Comment (mosaaleb) [inline app/views/events/show.html.erb]: - [ ] We need to hide the button if the user is already attending the event. Remember to use `helpers` to handle logic in views.

File: app/views/users/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,51 @@
+<h1><%= @user.name %></h1>
+<h2>Created events</h2>
+<ul>
+<% if @user.created_events.any?%>
```
Comment (mosaaleb) [inline app/views/users/show.html.erb]: - [ ] There shoud be absolutely NO logic in views. 
Rails has built-in mechanism to deal with view-related logic which is View Helpers (app/helpers/*_helper.rb). So the common convention is to move any logic in the views to the helpers. Please take a look at [this](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) for more information.

File: app/controllers/attendances_controller.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+class AttendancesController < ApplicationController
+  def create
+    @user = current_user
+    @event = Event.find(params[:id])
+    @attendance = Attendance.new(user: @user, event: @event)
+
+    if @attendance.valid?
+      @attendance.save
+      flash.now[:alert] = 'You signed up to this event successfully!'
+    else
```
Comment (mosaaleb) [inline app/controllers/attendances_controller.rb]: - [ ] Flash alert is not showing up. The right syntax is `flash[:notice]` or `flash[:alert]`

File: app/helpers/sessions_helper.rb [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+  end
+
+  # Logs out the current user.
+  def log_out
+    forget(current_user)
+    session.delete(:user_id)
+    @current_user = nil
+  end
+
+  # def require_login
```
Comment (mcrd25) [inline app/helpers/sessions_helper.rb]: It is required that you remove all commented out. 
It’s better to delete commented code. If you need to review a previous version of your code, use git history. Please check out this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/ link that lists it as a requirement.

File: app/helpers/sessions_helper.rb [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    end
+  end
+
+  # Returns true if the given user is the current user.
+  def current_user?(user)
+    user == current_user
+  end
+
+  # Returns true if the user is logged in, false otherwise.
+  def logged_in?
```
Comment (mcrd25) [inline app/helpers/sessions_helper.rb]: A method like this is used for views and can stay as a candidate for the helper file. But I noticed that you use it in controllers so what you can do is keep it as a application controller method and list it as a helper method there like following:
`helper_method :logged_in?`

File: app/helpers/sessions_helper.rb [hunk 8]
```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module SessionsHelper
```
Comment (mcrd25) [inline app/helpers/sessions_helper.rb]: You seem to have application logic inside of your helpers. Helper methods ought to be view specific; the fact that you have to import the helpers in Application Controller for you to have access to those methods in controllers while you do not even use it in your views reiterates this. The views on the other hand, automatically have access to helper methods. 
So, what you can do is add all methods which are not views specific to your application controller directly. What you can also do if your method list gets long, you may use concerns or a lib folder to separate methods by function, etc.

Basically helpers = views, controller = controller. Why write methods in helpers only to have to include it in controller after. Helpers are automatically accessible in views. there is a reason helpers is outside the context of controller folder. You can, however, use concerns in your controllers.

Read [this article](http://codefol.io/posts/Where-Do-I-Put-My-Code/) for a clearer explanation. 🙂 

Just to be clear:
Helper methods = view specific logic not controller logic

**Pull request merged**